File: wrappers/sbca-python-wrapper/sbca_wrapper/_commands/pool.py
from typing import Optional, Union

from .._command import LibindyCommand


class Pool:

    @staticmethod
    @LibindyCommand('indy_create_pool_ledger_config')
    async def create_pool_config(
            config_name: str,
            config: Optional[Union[dict, str]]
    ):
        """"""
        pass

    @staticmethod
    @LibindyCommand('indy_delete_pool_ledger_config')
    async def delete_pool_config(
            config_name: str
    ):
        """"""
        pass

    @staticmethod
    @LibindyCommand('indy_open_pool_ledger')
    async def open_pool_connection(
            config_name: str,
            config: Optional[Union[dict, str]]
    ) -> int:
        """"""
        pass

    @staticmethod
    @LibindyCommand('indy_close_pool_ledger')
    async def close_pool_connection(
            pool_handle: int
    ):
        """"""
        pass

    @staticmethod
    @LibindyCommand('indy_refresh_pool_ledger')
    async def refresh_local_pool_ledger(
            pool_handle: int
    ):
        """"""
        pass

    @staticmethod
    @LibindyCommand('indy_list_pools')
    async def list_local_pool_ledgers() -> list:
        """"""
        pass

    @staticmethod
    @LibindyCommand('indy_set_protocol_version',
                    protocol_version=lambda arg: arg)
    async def set_protocol_version(
            protocol_version: int
    ):
        """"""
        pass

File: wrappers/rust/indy-sys/src/pool.rs
use super::*;

use {CString, Error, CommandHandle, PoolHandle};

extern {

    pub fn indy_create_pool_ledger_config(command_handle: CommandHandle,
                                          config_name: CString,
                                          config: CString,
                                          cb: Option<ResponseEmptyCB>) -> Error;

    pub fn indy_open_pool_ledger(command_handle: CommandHandle,
                                 config_name: CString,
                                 config: CString,
                                 cb: Option<ResponseI32CB>) -> Error;

    pub fn indy_refresh_pool_ledger(command_handle: CommandHandle,
                                    handle: PoolHandle,
                                    cb: Option<ResponseEmptyCB>) -> Error;

    pub fn indy_list_pools(command_handle: CommandHandle,
                           cb: Option<ResponseStringCB>) -> Error;

    pub fn indy_close_pool_ledger(command_handle: CommandHandle,
                                  handle: PoolHandle,
                                  cb: Option<ResponseEmptyCB>) -> Error;

    pub fn indy_delete_pool_ledger_config(command_handle: CommandHandle,
                                          config_name: CString,
                                          cb: Option<ResponseEmptyCB>) -> Error;

    pub fn indy_set_protocol_version(command_handle: CommandHandle,
                                     protocol_version: usize,
                                     cb: Option<ResponseEmptyCB>) -> Error;
}

